pkg/rpc/client.go

```diff
@@ -48,6 +57,22 @@ func NewClient(connectionMode ConnectionMode, options ...rpcinterface.ClientOpti
 		config: cfg,
 	}
 
+	if ports.FullNode != 0 {
+		c.config.FullNode.RPCPort = ports.FullNode
+	}
+	if ports.Farmer != 0 {
+		c.config.Farmer.RPCPort = ports.Farmer
+	}
+	if ports.Harvester != 0 {
+		c.config.Harvester.RPCPort = ports.Harvester
+	}
+	if ports.Wallet != 0 {
+		c.config.Wallet.RPCPort = ports.Wallet
+	}
+	if ports.Crawler != 0 {
+		c.config.Seeder.CrawlerConfig.RPCPort = ports.Crawler
+	}
+
```

Comment:
Would rather do these as clientoptions - see for example: https://github.com/Chia-Network/go-chia-libs/blob/main/pkg/rpc/clientoptions.go#L32

Reply:
```NewClient(ConnectionModeHTTP, WithWalletPort(1234), WithNodePort(5678))```

Reply:
Good suggestion :+1: I'm not quite there yet, my priority for now is just getting the testing pattern nailed before getting to a better solution on client options for ports. I haven't ingested a _ton_ of the client code yet, but this might be more complicated because the current client options interact a layer lower into the actual client than the port confic IIRC. But yeah, I'll get here eventually :)

Reply:
Done`